src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -909,20 +911,31 @@ private byte[] GetAsciiEncoding(byte value)
             return encoding;
         }
 
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private void EnsureAsciiCacheInitialized()
```

Comment:
Can you do a perf test or this, especially for small strings?

I don't know if we should explicitly mark this method as no inlining since presumably, this should be a quick check and return.

Would it be better to move the quick check into the caller, or maybe, move the initialization for-loop into a non-inlinable method, but keep this method undecorated?

Reply:
I'll investigate this.

Reply:
I think it's better to handle this in a separate PR. So I'll remove this here, to keep this PR for the SSSE3-part only.

Reply:
-> https://github.com/dotnet/corefx/pull/42177